core-feature-pack/common/src/main/resources/content/bin/domain.sh

```diff
@@ -105,38 +105,6 @@ JVM_D32_OPTION=`echo $JAVA_OPTS | $GREP "\-d32"`
 test "x$JVM_D64_OPTION" != "x" && JVM_OPTVERSION="-d64 $JVM_OPTVERSION"
 test "x$JVM_D32_OPTION" != "x" && JVM_OPTVERSION="-d32 $JVM_OPTVERSION"
 
-# If -server not set in JAVA_OPTS, set it, if supported
-SERVER_SET=`echo $JAVA_OPTS | $GREP "\-server"`
-if [ "x$SERVER_SET" = "x" ]; then
-
-    # Check for SUN(tm) JVM w/ HotSpot support
-    if [ "x$HAS_HOTSPOT" = "x" ]; then
-        HAS_HOTSPOT=`"$JAVA" $JVM_OPTVERSION -version 2>&1 | $GREP -i HotSpot`
-    fi
-
-    # Check for OpenJDK JVM w/server support
-    if [ "x$HAS_OPENJDK" = "x" ]; then
-        HAS_OPENJDK=`"$JAVA" $JVM_OPTVERSION 2>&1 | $GREP -i OpenJDK`
-    fi
-
-    # Check for IBM JVM w/server support
-    if [ "x$HAS_IBM" = "x" ]; then
-        HAS_IBM=`"$JAVA" $JVM_OPTVERSION 2>&1 | $GREP -i "IBM J9"`
-    fi
-
-    # Enable -server if we have Hotspot or OpenJDK, unless we can't
-    if [ "x$HAS_HOTSPOT" != "x" -o "x$HAS_OPENJDK" != "x" -o "x$HAS_IBM" != "x" ]; then
-        # MacOS does not support -server flag
-        if [ "$darwin" != "true" ]; then
-            PROCESS_CONTROLLER_JAVA_OPTS="-server $PROCESS_CONTROLLER_JAVA_OPTS"
-            HOST_CONTROLLER_JAVA_OPTS="-server $HOST_CONTROLLER_JAVA_OPTS"
-            JVM_OPTVERSION="-server $JVM_OPTVERSION"
-        fi
-    fi
-else
-    JVM_OPTVERSION="-server $JVM_OPTVERSION"
-fi
-
```

Comment:
I think it is harmless, the code is prepared to add the "-server" setting automatically if it is supported. The same is applicable to the `standalone.bat` with -client and -server. 

@ThePrez I could be missing something, what's the reason for the removal?

Reply:
This code in `domain.sh` also breaks with system Java on IBM i (domain mode is likely very uncommon on IBM i so I didn't do proper testing with it). 

Rather than cleaning it up so it works, removal seemed to make the most sense, given the comments on [Zulip](https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/JVM.20-server.20option). The `-server` option has lost its relevancy over time.

Reply:
> Ok. I thought it was not going to fail since the above code only will add the "-server" option for HotSpot/OpenJDK and IBM J9

FWIW, System Java on this platform is IBM J9, but not all flavors of IBM J9 support `-server` 😖

Reply:
Ok, I thought we were going to have a different version, then removing that completely is safest for being able to use it on IBM i.